myhoard/binary_io_slice.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2024 Aiven, Helsinki, Finland. https://aiven.io/
+from types import TracebackType
+from typing import BinaryIO, Type, Iterable, Iterator
+
+
+class MethodNotSupportedError(Exception):
+    pass
+
+
+class BinaryIOSlice(BinaryIO):
+    def __init__(self, max_file_size: int, stream: BinaryIO):
+        super().__init__()
+        self._size_remaining = max_file_size
+        self.stream = stream
+
+    def read(self, __n: int = ...) -> bytes:
+        to_read = min(__n, self._size_remaining)
```

Comment:
I am wondering if it does what we want to in case if `__n = -1`, the `min` will return `-1` in this case, but we still want to read up to `self._size_remaining` ?

Reply:
I can just add add a check for >= 0 to solve that? Is -1 even supported by the IO classes?

Reply:
Seems to be supported, I'll probably just raise a `ValueError` in that case:
```
$ echo "asdf" > foo.txt
$ python
Python [...]
>>> with open("foo.txt", "r") as f:
...    print(f.read(-1))
... 
asdf

>>>
```

Reply:
It should mean read until EOF if I am not mistaken.

Reply:
Rohmu will not use that but I've changed it to this, seems to be what a normal `read` does:
```
        if __n < 0:
            to_read = self._size_remaining
        else:
            to_read = min(__n, self._size_remaining)
```